Extensions/TextObject/TextObject.cpp

```diff
@@ -27,52 +25,112 @@ TextObject::TextObject()
       bold(false),
       italic(false),
       underlined(false),
-      colorR(0),
-      colorG(0),
-      colorB(0),
-      textAlignment("left")
+      color("0;0;0"),
+      textAlignment("left"),
+      isOutlineEnabled(false),
+      outlineThickness(2),
+      outlineColor("255;255;255"),
+      isShadowEnabled(false),
+      shadowColor("0;0;0"),
+      shadowOpacity(127),
+      shadowAngle(90),
+      shadowDistance(4),
+      shadowBlurRadius(2)
 {
 }
 
 TextObject::~TextObject(){};
 
 void TextObject::DoUnserializeFrom(gd::Project& project,
                                    const gd::SerializerElement& element) {
-  SetString(element.GetChild("string", 0, "String").GetValue().GetString());
-  SetFontName(element.GetChild("font", 0, "Font").GetValue().GetString());
-  SetTextAlignment(element.GetChild("textAlignment").GetValue().GetString());
-  SetCharacterSize(element.GetChild("characterSize", 0, "CharacterSize")
+
+  // Compatibility with 5.3.187 or older releases.
+  bool isLegacy = !element.HasChild("content");
+  auto &content = isLegacy ? element : element.GetChild("content");
+
+  SetFontName(content.GetChild("font", 0, "Font").GetValue().GetString());
+  SetTextAlignment(content.GetChild("textAlignment").GetValue().GetString());
+  SetCharacterSize(content.GetChild("characterSize", 0, "CharacterSize")
                        .GetValue()
                        .GetInt());
-  SetColor(element.GetChild("color", 0, "Color").GetIntAttribute("r", 255),
-           element.GetChild("color", 0, "Color").GetIntAttribute("g", 255),
-           element.GetChild("color", 0, "Color").GetIntAttribute("b", 255));
+  smoothed = content.GetBoolAttribute("smoothed");
+  bold = content.GetBoolAttribute("bold");
+  italic = content.GetBoolAttribute("italic");
+  underlined = content.GetBoolAttribute("underlined");
 
-  smoothed = element.GetBoolAttribute("smoothed");
-  bold = element.GetBoolAttribute("bold");
-  italic = element.GetBoolAttribute("italic");
-  underlined = element.GetBoolAttribute("underlined");
+  if (isLegacy) {
+    // Compatibility with 5.3.187 or older releases.
+    SetText(content.GetChild("string", 0, "String").GetValue().GetString());
+    SetColor(
+        gd::String::From(
+            content.GetChild("color", 0, "Color").GetIntAttribute("r", 255)) +
+        ";" +
+        gd::String::From(
+            content.GetChild("color", 0, "Color").GetIntAttribute("g", 255)) +
+        ";" +
+        gd::String::From(
+            content.GetChild("color", 0, "Color").GetIntAttribute("b", 255)));
+  } else {
```

Comment:
Might be a good idea to reset shadow/outline to specific values?

Reply:
I think it's already done by the constructor.